tests/test_options.py

```diff
@@ -212,3 +264,365 @@ def test_as_dict():
+        assert value == 'mymodel'
+
+
+def test_lock():
+    reset_options_instance()
+    source = {
+        'dataset': 123,
+        'model': {
+            'criterion': 'I am a criterion',
+            'network': 'I am a network',
+        },
+    }
+    Options(source, run_parser=False)
+    assert Options().options == source
+    Options().unlock()
+    assert not Options().options.islocked()
+    assert not Options().options['model'].islocked()
+    Options().lock()
+    assert Options().options.islocked()
+    assert Options().options['model'].islocked()
+
+
+def test_unlock():
+    reset_options_instance()
+    source = {
+        'dataset': 123,
+        'model': {
+            'criterion': 'I am a criterion',
+            'network': 'I am a network',
+        },
+    }
+    Options(source, run_parser=False)
+    assert Options().options == source
+    Options().lock()
+    assert Options().options.islocked()
+    assert Options().options['model'].islocked()
+    Options().unlock()
+    assert not Options().options.islocked()
+    assert not Options().options['model'].islocked()
+
+
+def test_lock_setitem():
+    reset_options_instance()
+    source = {
+        'dataset': 123,
+        'model': {
+            'criterion': 'I am a criterion',
+            'network': 'I am a network',
+        },
+    }
+    Options(source, run_parser=False)
+    assert Options().options == source
+    Options().lock()
+    with pytest.raises(PermissionError):
+        Options()['dataset'] = 421
+
+
+def test_str_to_bool_yes():
+    reset_options_instance()
+    source = {'abc': 123}
+    Options(source, run_parser=False)
+    assert Options().str_to_bool('yes')
+    assert Options().str_to_bool('Yes')
+    assert Options().str_to_bool('YES')
+
+
+def test_str_to_bool_true():
+    reset_options_instance()
+    source = {'abc': 123}
+    Options(source, run_parser=False)
+    assert Options().str_to_bool('true')
+    assert Options().str_to_bool('True')
+    assert Options().str_to_bool('TRUE')
+
+
+def test_str_to_bool_no():
+    reset_options_instance()
+    source = {'abc': 123}
+    Options(source, run_parser=False)
+    assert not Options().str_to_bool('no')
+    assert not Options().str_to_bool('No')
+    assert not Options().str_to_bool('NO')
+
+
+def test_str_to_bool_false():
+    reset_options_instance()
+    source = {'abc': 123}
+    Options(source, run_parser=False)
+    assert not Options().str_to_bool('false')
+    assert not Options().str_to_bool('False')
+    assert not Options().str_to_bool('FALSE')
+
+
+def test_str_to_bool_incorrect():
+    reset_options_instance()
+    source = {'abc': 123}
+    Options(source, run_parser=False)
+    with pytest.raises(argparse.ArgumentTypeError):
+        Options().str_to_bool('incorrect')
+
+
+def test_str():
+    reset_options_instance()
+    source = {'abc': 123}
+    Options(source, run_parser=False)
+    assert Options().options == source
+    assert Options().__str__() == '{\n  "abc": 123\n}'
```

Comment:
This is sort of complicated... If we change the serialization pattern, the `\n` might not exist, for example, and this whole thing will fail. I suppose we can let this test as it is, and fix it later if we ever change the serialization pattern